classes/class-p4-taxonomy-image.php

```diff
@@ -42,33 +42,36 @@ private function hooks() {
 		 * @param WP_Term $wp_tag The object passed to the callback when on Edit Tag page.
 		 */
 		public function add_taxonomy_form_fields( $wp_tag ) {
-			if ( isset( $wp_tag ) && $wp_tag instanceof WP_Term ) { ?>
-				<tr class="form-field term-image-wrap">
+			if ( isset( $wp_tag ) && $wp_tag instanceof WP_Term ) {
+				$attachment_id  = get_term_meta( $wp_tag->term_id, 'tag_attachment_id', true );
+				$attachment_url = get_term_meta( $wp_tag->term_id, 'tag_attachment', true ); ?>
```

Comment:
Anything grabbing an image src url should ultimately go through wp_get_attachment_image_src on render which is used for all sorts of filtering - dynamic generation means less issues modifying permastructs, integrating off loaded image serving, on the fly resizing etc. Object, fragment and static page caching should ultimately dictate the site's performance.